File: Brute Force/Climbing_Stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int arr[n+1];
        arr[0] = 1;
        arr[1] = 2;

        for(int i = 1; i<n-1; i++){
            arr[i+1] = arr[i] + arr[i-1]; 
        }
        return arr[n-1];
    }
};

File: Dynamic Programming/Climbing_Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1
        
        if n == 2:
            return 2

        prev, nxt = 1, 2

        for i in range(2, n):
            prev, nxt = nxt, prev + nxt

        return nxt